src/Arcus.Scripting.IntegrationAccount/Scripts/Set-AzIntegrationAccountAssemblies.ps1

```diff
@@ -0,0 +1,73 @@
+    [Parameter(Mandatory = $false)][string] $ArtifactsPrefix = ''
+)
+
+if ($AssemblyFilePath -ne '' -and $AssembliesFolder -ne '') {
+    throw "Either the file path of a specific assembly or the file path of a folder containing multiple assemblies is required, e.g.: -AssemblyFilePath 'C:\Assemblies\assembly.dll' or -AssembliesFolder 'C:\Assemblies'"
+}
+
+function UploadAssembly {
+    param
+    (
+        [System.IO.FileInfo][parameter(Mandatory = $true)]$Assembly
+    )
+
+    $assemblyName = $Assembly.BaseName
+    if ($ArtifactsPrefix -ne '') {
+        $assemblyName = $ArtifactsPrefix + $assemblyName
+    }
+    Write-Host "Uploading assembly '$assemblyName' into the Integration Account '$Name'."
+
+    $existingAssembly = $null
+    try {
+        Write-Verbose "Checking if the assembly '$assemblyName' already exists in the Integration Account '$Name'."
+        $existingAssembly = Get-AzIntegrationAccountAssembly -ResourceGroupName $ResourceGroupName -IntegrationAccount $Name -Name $assemblyName -ErrorAction Stop
+    }
+    catch {
+        if ($_.Exception.Message.Contains('could not be found')) {
+            Write-Verbose "No assembly '$assemblyName' could not be found in Azure Integration Account '$Name'."
+        }
+        else {
+            throw $_.Exception
+        }
+    }
+        
+    try {
+        if ($existingAssembly -eq $null) {
+            Write-Verbose "Creating assembly '$assemblyName' in Azure Integration Account '$Name'"
+            $createdAssembly = New-AzIntegrationAccountAssembly -ResourceGroupName $ResourceGroupName -IntegrationAccount $Name -Name $assemblyName -AssemblyFilePath $Assembly.FullName -ErrorAction Stop
+            Write-Verbose ($createdAssembly | Format-List -Force | Out-String)
+        }
+        else {
+            Write-Verbose "Updating assembly '$assemblyName' in Azure Integration Account '$Name'"
+            $updatedAssembly = Set-AzIntegrationAccountAssembly -ResourceGroupName $ResourceGroupName -IntegrationAccount $Name -Name $assemblyName -AssemblyFilePath $Assembly.FullName -ErrorAction Stop
+            Write-Verbose ($updatedAssembly | Format-List -Force | Out-String)
+        }
+        Write-Host "Assembly '$assemblyName' has been uploaded into the Azure Integration Account '$Name'."
+    }
+    catch {
+        Write-Error "Failed to upload assembly '$assemblyName' in Azure Integration Account '$Name': '$($_.Exception.Message)_'"
+    }
+}
+
+# Verify if Integration Account can be found based on the given information
+$integrationAccount = Get-AzIntegrationAccount -ResourceGroupName $ResourceGroupName -Name $Name -ErrorAction SilentlyContinue
+if ($integrationAccount -eq $null) {
+    Write-Error "Unable to find the Azure Integration Account with name '$Name' in resource group '$ResourceGroupName'"
```

Comment:
Shouldn't this be `throw` as we can't do anything beyond this point?

Reply:
This is handled by the if statement, the script will only execute the actual upload if the integration account is found.

No problem to change it, but it's the same in the already existing scripts to upload schemas and maps, so would also need to change that to keep it consistent.

Reply:
Did this, to make to outputted error a bit more user-friendly/readable.
Also, what Pim said :)

Reply:
I guess it's a matter of choice and usability. Not sure if consumers using this, would like to fail-fast instead of having a more obscure error further up the line; or would like to run this as 'try to upload it, but no worries if it fails'. Not sure 😕

Reply:
Did a quick test using both `Write-Error` or `throw` in a DevOps build pipeline, the result is not that much different, both log the necessary error information and both mark the the run as failed.

I would say keep it as is. Whether it is `Write-Error` or `throw` it is the last statement in the script, so both versions fail equally as fast 😉 Also not sure if we break anything regarding backwards compatibility if we move to `throw`?

Reply:
Is good to leave it, I guess. The places where it's necessary to block the script are already using `throw`, so I guess it's good like it is. We can change it if threre's need for it. 👍